releases/fuji/compose-files/docker-compose-fuji-1.0.0.yml

```diff
@@ -0,0 +1,579 @@
+#  * updated: October 8, 2019
+#  *******************************************************************************/
+
+version: '3'
+volumes:
+  db-data:
+  log-data:
+  consul-config:
+  consul-data:
+  portainer_data:
+  vault-config:
+  vault-file:
+  vault-logs:
+  secrets-setup-cache:
+
+services:
+  volume:
+    image: edgexfoundry/docker-edgex-volume:1.0.0
+    container_name: edgex-files
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-files
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+      - secrets-setup-cache:/etc/edgex/pki
+
+  consul:
+    image: consul:1.3.1
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+      - "8600:8600"
+    container_name: edgex-core-consul
+    hostname: edgex-core-consul
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-consul
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+
+  config-seed:
+    image: edgexfoundry/docker-core-config-seed-go:1.0.0
+    container_name: edgex-config-seed
+    hostname: edgex-core-config-seed
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-config-seed
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+      - consul
+
+  vault:
+    image: edgexfoundry/docker-edgex-vault:1.1.0-dev
+    container_name: edgex-vault
+    hostname: edgex-vault
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-vault
+    ports:
+      - "8200:8200"
+    cap_add:
+      - "IPC_LOCK"
+    environment:
+      - VAULT_ADDR=https://edgex-vault:8200
+      - |
+        VAULT_LOCAL_CONFIG=
+          listener "tcp" { 
+              tls_disable = "0" 
+              cluster_address = "edgex-vault:8201" 
+              tls_min_version = "tls12" 
+              tls_client_ca_file ="/run/edgex/secrets/edgex-vault/ca.pem" 
+              tls_cert_file ="/run/edgex/secrets/edgex-vault/server.crt" 
+              tls_key_file = "/run/edgex/secrets/edgex-vault/server.key" 
+              tls_perfer_server_cipher_suites = true 
+          } 
+          backend "consul" { 
+              path = "vault/" 
+              address = "edgex-core-consul:8500" 
+              scheme = "http" 
+              redirect_addr = "https://edgex-vault:8200" 
+              cluster_addr = "https://edgex-vault:8201" 
+          } 
+          default_lease_ttl = "168h" 
+          max_lease_ttl = "720h"
+      - VAULT_CONFIG_DIR=/vault/config
+      - VAULT_UI=true
+    tmpfs:
+      - /run
+    command: >
+      /usr/bin/dumb-init -- /bin/sh -c 
+      "./security-secrets-setup generate ; 
+      chown -Rh 100:1000 /run/edgex/secrets/edgex-vault/ ;
```

Comment:
why owner `100` and group `1000`?

Reply:
Because we want to make sure vault:vault usergroup and userid has the access to this deploy directory where Vault's TLS asset resides.  The Vault's docker image creates that vault:vault and we are inherited it from that docker image so the owner and the group is always 100:1000 in the docker container.

Reply:
ah okay, I wasn't aware that the vault image used those specific uids, this is fine then